### PR TITLE
Support Snowflake key pair auth for source connector

### PR DESCRIPTION
**Pull request by aharbunou-branch**
## What
Addressing existed issue https://github.com/airbytehq/airbyte/issues/14615

## How
I used destination Snowflake connector implementation as a reference to support key pair auth for source connector.
https://github.com/airbytehq/airbyte/tree/master/airbyte-integrations/connectors/destination-snowflake

## Review guide
1. `y.json`
2. `x.java`

## User Impact
Users will be able to connect to Snowflake via Key Pair auth method.

Screenshot 2024-06-24 at 6 50 26 PM

There is no breaking changes.

## Can this PR be safely reverted and rolled back?
- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by evantahler — APPROVED** [review]
Looks safe enough to me, @marcosmarxm.  If you can confirm this works in our sandbox, 👍 from me!

File: airbyte-integrations/connectors/source-snowflake/src/main/resources/spec.json [hunk 1]
```diff
@@ -52,11 +52,46 @@
...
+                "description": "The username you created to allow Airbyte to access the database.",
+                "examples": ["AIRBYTE_USER"],
+                "type": "string",
+                "title": "Username",
+                "order": 1
+              },
+              "private_key": {
+                "type": "string",
+                "title": "Private Key",
+                "description": "RSA Private key to use for Snowflake connection. See the <a href=\"https://docs.airbyte.com/integrations/destinations/snowflake\">docs</a> for more information on how to obtain this key.",
```
Comment (theyueli) [inline airbyte-integrations/connectors/source-snowflake/src/main/resources/spec.json:76]: To be more accurate, please use this link

https://docs.airbyte.com/integrations/destinations/snowflake#key-pair-authentication

@evantahler I think we need to copy that particular key pair setup section of the doc into the Snowflake source documentation page as well, and then link it to the source page instead. @aharbunou-branch would you like to help update the doc in the PR too?

Comment (aharbunou-branch) [reply airbyte-integrations/connectors/source-snowflake/src/main/resources/spec.json]: @theyueli I've updated documentation for source snowflake connector. I also updated changelog and bumped connector version. Let me know if it looks good.

**Comment by marcosmarxm** [comment]
@aharbunou-branch running CI tests let's wait for the results.

File: airbyte-integrations/connectors/source-snowflake/src/main/resources/spec.json (hunk 1 above)
Comment (theyueli) [reply airbyte-integrations/connectors/source-snowflake/src/main/resources/spec.json]: Thank you! looks good!

**Review by theyueli — APPROVED** [review]
Thanks for your contribution!

**Comment by aharbunou-branch** [comment]
> @aharbunou-branch running CI tests let's wait for the results.

@marcosmarxm does CI looks good? Is there anything else I need to do before the PR can be merged?

**Comment by marcosmarxm** [comment]
@aharbunou-branch ask you to not update the branch as it will cancel the CI and will require me to start it again.

**Comment by therji-motif** [comment]
Do we know when this will be deployed?

**Pull request merged**